beacon_node/network/src/sync/block_processor.rs

```diff
@@ -174,6 +203,67 @@ fn process_batch<T: BeaconChainTypes>(
     Ok(())
 }
 
+/// Handles the parent lookup processing given a list of blocks.
+fn parent_lookup<T: BeaconChainTypes>(
```

Comment:
I was thinking we could remove this all-together. Just have a single `process_blocks` function, which also contains the `std::thread::spawn`. 
Then in the above match, we just call `process_block` for both types of process-id. 
The logic is slightly different between the two. The logic in the current `process_batch` I think is the one we should go with. 
The main difference (I think) between the two, is that one consumes the blocks (parent_lookup) whereas the other clones the blocks. 
I'm currently leaning towards just using the `process_batch` although maybe rename it, and keep the cloning logic. The parent lookup is less likely and should have a smaller list of blocks so the cloning will have minimal overhead and we gain having a single set of logic that governs how blocks are processed.